web/src/modules/administration/modules/injest-data/store/index.ts

```diff
@@ -0,0 +1,111 @@
+  state: (): AdminState => ({
+    dataIngestionSources: [],
+    isLoading: false,
+    selectedDataIngestionSourceId: undefined,
+    selectedDataIngestionFile: undefined,
+  }),
+  getters: {
+    dataIngestionSourceCount(state) {
+      if (state && state.dataIngestionSources) return state.dataIngestionSources.length;
+      return 0;
+    },
+  },
+  actions: {
+    async getAllDataIngestionSources() {
+      this.isLoading = true;
+      let api = useApiStore();
+      await api
+        .secureCall("get", DATA_INGESTION_SOURCE_URL)
+        .then((resp) => {
+          this.dataIngestionSources = resp.dataIngestionSources;
+        })
+        .finally(() => {
+          this.isLoading = false;
+        });
+    },
+    selectDataIngestionSourceId(dataIngestionSourceId: number) {
+      this.selectedDataIngestionSourceId = dataIngestionSourceId;
+    },
+    unselectDataIngestionSourceId() {
+      this.selectedDataIngestionSourceId = undefined;
+    },
+    selectDataIngestionFile(dataIngestionFile: File) {
+      this.selectedDataIngestionFile = dataIngestionFile;
+    },
+    unselectDataIngestionFile() {
+      this.selectedDataIngestionFile = undefined;
+    },
+    async addDataIngestion(userId: number) {
+      const api = useApiStore();
+      const notify = useNotificationStore();
+
+      // Guards
+      if (!this.selectedDataIngestionSourceId) {
+        return notify.notify({
+          text: "Please select a data ingestion source first",
+          variant: "error",
+        });
+      }
+      if (!this.selectedDataIngestionFile) {
+        return notify.notify({ text: "Please choose a CSV file to upload", variant: "error" });
+      }
+
+      this.isLoading = true;
+      try {
+        const form = new FormData();
+        form.append("csvFile", this.selectedDataIngestionFile);
+        form.append("source_id", this.selectedDataIngestionSourceId.toString());
+        form.append("user_id", userId.toString());
+
+        const result = await api.secureUpload("post", DATA_INGESTION_URL, form);
+
+        if (result.error) {
+          notify.notify({ text: String(result.error), variant: "error" });
+        } else {
+          notify.notify({ text: "Data Ingestion File added successfully", variant: "success" });
+          this.unselectDataIngestionSourceId();
+          this.unselectDataIngestionFile();
+        }
+
+        return result;
+      } catch (err: any) {
+        console.error("Data ingestion failed", err);
+        notify.notify({
+          text: err.message || "Upload failed. Please try again.",
+          variant: "error",
+        });
+        throw err;
+      } finally {
+        this.isLoading = false;
+      }
+    },
+  },
+});
+
+export interface DataIngestionSource {
+  id: number;
+  source_name: string;
+  description?: string;
+  identifier_column_name: string;
+  column_count: number;
+  source_attribute_to_transform?: string;
+  target_attribute_to_transform?: string;
+}
```

Comment:
Prefer putting type definitions in the front-end "api" layer, or at the top of the file.
e.g. https://github.com/icefoganalytics/wrap/blob/9da5e31931351ce911d441556caa388d4dc2ac35/web/src/api/delegations-api.ts#L9